include/irr/asset/IMeshPacker.h

```diff
@@ -21,13 +18,96 @@ template <typename MeshBufferType>
 class IMeshPacker
 {
 public:
-    virtual std::optional<std::pair<MeshBufferType* ,DrawElementsIndirectCommand_t>> packMeshes(const core::vector<MeshBufferType*>& meshBuffers) = 0;
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                    = 2147483648ull; /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                   = 4294967296ull; /*  4GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes        = 33554432ull;   /* 32MB*/
+        size_t MDIDataSizeInBytes                               = 2147483648ull; /*  2GB*/
```

Comment:
If you have 2GB of `uint16_t` indices, then you have 341M triangles, if you split meshbuffers into batches of 32-1024 tris, then your max MDI count is 10.6M, given that each MDI struct is 20 bytes... works out to 214MB max